# KeyError 

# This occurs when a dictionary does not have a specific key:

d = {}
d["foo"]
# KeyError: 'foo'